### DISCUSSION TITLE
Any pending changes before tagging v3.2.0?

### DISCUSSION THREAD

**Discussion by robgjansen** [shadow/shadow#3340] [General]
@sporksmith @stevenengler do we have any pending PRs or other local changes that we ought to merge before tagging a new minor release (v3.2.0)? (sporksmith requested a new minor version, I think because `main` contains some fixes useful for the arti CI that are not in the latest release.)

**Comment by sporksmith** [comment]
Ready here :)

**Reply by stevenengler** [reply]
Also good with me. One thing we haven't done in a while is update shadow's dependencies (we've done a patch version update recently with `cargo update` but not a major/minor version update), but I don't think that needs to hold back a new shadow release. Also maybe good to mention ppopth and jtracey in the release notes for the netlink, epoll, and doc improvements since those probably make up a large chunk of the added lines in this release.

**Reply by robgjansen** [reply]
Yeah, the CI workflow was broken, but I recently fixed that and now ran it manually:
https://github.com/shadow/shadow/issues/3341

I'm inclined to run those maintenance tasks in #3341 before the new release, assuming there is not a super-urgent need for a release.

**Reply by sporksmith** [reply]
Sounds good

**Reply by robgjansen** [reply]
I updated the versions of the rust deps, and it (perhaps expectedly) caused a bunch of errors. I started fixing some of the easy ones, but it would be nice to have help of the more complicated ones.

I pushed a WIP in https://github.com/shadow/shadow/pull/3343, feel free to push new fixup commits to that branch. I probably won't have more time to spend on it before the weekend.

**Reply by stevenengler** [reply]
> I updated the versions of the rust deps, and it (perhaps expectedly) caused a bunch of errors. I started fixing some of the easy ones, but it would be nice to have help of the more complicated ones.

I'll just mention that we should stick to an old version of nix in the tests crate for now, unless someone feels like tackling that issue:

https://github.com/shadow/shadow/issues/3265

https://github.com/shadow/shadow/pull/3264#issuecomment-1873495880

It will be a couple of days before I have a working computer again. So if no one gets to it before then, I can probably take a look at fixing the errors this weekend.

**Reply by sporksmith** [reply]
I *think* we have a rough consensus that we'd prefer not to depend on nix. I filed https://github.com/shadow/shadow/issues/3345 as a tracking issue for it.

I have a PR in progress to migrate some of our usage to rustix. I don't know that it's worth holding the release up on, but I figured it was at least worth doing a bit of a trial run of some of the migration. I think holding back `nix` from updating for this release would be fine. (And maybe preferable to dropping a major refactor bomb before cutting the release)

**Reply by robgjansen** [reply]
> (And maybe preferable to dropping a major refactor bomb before cutting the release)

I suppose the same could be said about updating the dependencies too. I've been running many shadow sims over the last months with the current deps and things have been stable, so perhaps I should cut the release now _before_ finishing #3343.

**Reply by sporksmith** [reply]
Sounds good to me

**Reply by robgjansen** [reply]
Done! https://github.com/shadow/shadow/releases/tag/v3.2.0

**Reply by stevenengler** [reply]
> It will be a couple of days before I have a working computer again. So if no one gets to it before then, I can probably take a look at fixing the errors this weekend.

Since it seems like most (all?) of the errors are due to nix, and many of those nix errors conflict with (and will be replaced by) the changes in #3346, I'm leaving #3343 alone for now.

**Comment by sporksmith** [comment]
Actually I think I just found a bug that'd be nice to squeak in a fix for before release.

`posix_spawn` is one of the few libc functions that returns errors directly in its return code instead of through errno. I think this  has caused me to go on a bit of a goose chase debugging an error further down the line (when we try to use pidfd_open on an invalid pid as a result...)

**Reply by sporksmith** [reply]
This fix is now merged.
